p4-16/spec/P4-16-spec.mdk

```diff
@@ -4888,13 +4888,15 @@ extern packet_in {
 To extract data from a packet represented by an argument `b` with
 type `packet_in`, a parser invokes the `extract` methods of `b`.
 There are two variants of the `extract` method: a one-argument
-variant for extracting fixed-size headers, and a two-argument variant
+variant for extracting structs or fixed-size headers, and a two-argument variant
 for extracting variable-sized headers. Because these operations can
 cause runtime verification failures (see below), these methods can
 only be executed within parsers.
 
-When extracting data into a bit-string or integer, the first packet
-bit is extracted to the most significant bit of the integer.
+When extracting data into a bit-string or integer field of a header,
```

Comment:
The struct itself won't tell you whether the extract has failed - it has no valid bit.

Reply:
For the use cases in mind, I don't think that is an issue.